docs/src/guide/staking/index.md

```diff
@@ -0,0 +1,136 @@
+      assume(tokenAmt >= 1, 'Must stake at least 1 token');
+  },
+  (tokenAmt) => {
+      //looks unintuitive, but this means the user will
+      // pay 0 algos, and pay tokenAmt in rewardTokens
+      return [0, [tokenAmt, stakeToken]]
+  },
+  (tokenAmt, apiReturn) => {
+      require(tokenAmt >= 1, 'Must stake at least 1 token');
+
+      userLocalState[this] = {
+          stake: tokenAmt,
+          pendingRewards: 0,
+      }
+      // return to the API caller
+      apiReturn(true)
+  }
+)
+```
+
+## Unstaking
+
+Unstaking is similar in structure.
+
+``` rsh
+.api(User.unstakeAndClaim,
+  () => {
+      // No rewards left in the contract? Sry!
+      assume(balance(rewardToken) > 0, 'No rewards left')
+  },
+  () => {
+      // transfer nothing to the contract
+      return [0, [0, stakeToken]]
+  },
+  (apiReturn) => {
+      require(balance(rewardToken) > 0, 'No rewards left')
+      const userState = userLocalState[this];
+      // pay acrued rewards
+      transfer([[userState.pendingrewards, rewardToken]]).to(this);
+
+      userLocalState[this] = {
+          stake: 0,
+          pendingRewards: 0,
+      }
+      // return to the API caller
+      apiReturn(true)
+      // update global state
+      return [
+          rewardSupply - userState.pendingRewards
+      ]
+  }
+)
+```
+
+## Connect to View
+
+Now for the cool part: Use your contract to hook to your website.
+
+``` rsh
+// connect to the contract
+const application = account.contract(backend, APPLICATION_ID);
+// defines all User api calls (stakeTokens, unstaketokens)
+const userApi = contract.a.User;
+// this is a view; read the reach docs at docs.reach.sh
+const userView = contract.v.read;
+
+useEffect(async() => {
+    // get users local state from application
+    const userState = await userView.read(account.addr);
+    setUserState(userState);
+}, []);
+
+<div>
+  {/* On button clicks, these functions will trigger MyAlgo */}
+  <button onClick={async () => await userApi.stakeTokens(amt)}>
+    Stake Tokens
+  </button>
+
+  <button onClick ={async () => await userApi.unstakeTokens(amt)}>
+    Unstake Tokens + {userState.pendingRewards} rewards
+  </button>
+</div>
+```
+
+## Conclusion
+
+This demonstrates the essential (but not quite complete) elements of implementing a contract in Reach.
+There's no magic involved, but it definitely feels like there is when you're building a Reach application. 
+As a developer, you don't need to think about forming transaction groups, SDK calls, or opting in. 
+That's right, Reach handles ASA and Application opt-ins. You focus on the business logic of your DApp and plug into the consensus network of your choice.
+
+In addition to all of this, Reach also provides formal verification and cross chain compatability. 
+At the time of this writing, Reach currently supports contract deployment on EVM (Ethereum Virtual Machine), AVM (Algorand Virtual Machine), and Conflux consensus networks.
```

Comment:
Worried about this line getting out of date over time

Reply:
Removed the final paragraph to keep the guide focused on staking.